### PR TITLE
refactor: rename function prefix mb to the more accurate utf_cp

### LINKED ISSUE #7401: Deprecate mb_* aliases
Since they reference both `utf_*` and `utfc_*` functions which have different semantics using them is going to be misleading and result in bugs like vim/vim#1827. Removing `utfc_` from aliases on the other hand is not a good idea either: this will lead to bugs arising from merging Vim patches without adjusting according to the semantic change.

Ref #7234.

### PR DESCRIPTION
**Pull request by dundargoc**
The "cp" stands for codepoint.

Closes https://github.com/neovim/neovim/issues/7401

### CODE REVIEW

File: src/nvim/mbyte.c [hunk 1]
```diff
@@ -1840,7 +1840,7 @@ int mb_off_next(const char_u *base, const char_u *p)
 /// Return the offset from "p" to the last byte of the character it points
 /// into.  Can start anywhere in a stream of bytes.
 /// Composing characters are not included.
```
Comment (clason) [inline src/nvim/mbyte.c:1842]:
```suggestion
/// Return the offset from `p_in` to the last byte of the codepoint it points
/// to.  Can start anywhere in a stream of bytes.
/// Note: Unlike `utf_tail_off`, individual codepoints of composing characters 
/// are counted separately.
```
(and similarly for `utf_cp_head_off`, which I can't comment on).

File: src/nvim/mbyte.c [hunk 2]
```diff
@@ -1837,10 +1837,11 @@ int mb_off_next(const char_u *base, const char_u *p)
   return i;
 }
 
-/// Return the offset from "p" to the last byte of the character it points
-/// into.  Can start anywhere in a stream of bytes.
-/// Composing characters are not included.
-int mb_tail_off(const char *base, const char *p_in)
+/// Return the offset from `p_in` to the last byte of the codepoint it points
+/// to.  Can start anywhere in a stream of bytes.
+/// Note: Unlike `utf_tail_off`, counts individual codepoints of composed characters
```
Comment (zeertzjq) [inline src/nvim/mbyte.c:1842]: `utf_tail_off` does not exist

Comment (clason) [reply src/nvim/mbyte.c]: 🤦

File: src/nvim/mbyte.c [hunk 3]
```diff
@@ -1866,15 +1865,16 @@ int mb_tail_off(const char *base, const char *p_in)
   return i;
 }
 
-/// Return the offset from "p" to the first byte of the character it points
-/// into. Can start anywhere in a stream of bytes.
-/// Unlike utf_head_off() this doesn't include composing characters and returns a negative value.
+/// Return the offset from "p" to the first byte of the codepoint it points
+/// to. Can start anywhere in a stream of bytes.
+/// Note: Unlike `utf_head_off`, this counts individual codepoints of composed characters
+/// separately.
```
Comment (clason) [inline src/nvim/mbyte.c:1871]:
```suggestion
/// separately and returns a negative offset.
```

File: src/nvim/mbyte.c [hunk 4]
```diff
@@ -1837,10 +1837,9 @@ int mb_off_next(const char_u *base, const char_u *p)
   return i;
 }
 
-/// Return the offset from "p" to the last byte of the character it points
-/// into.  Can start anywhere in a stream of bytes.
-/// Composing characters are not included.
-int mb_tail_off(const char *base, const char *p_in)
+/// Return the offset from `p_in` to the last byte of the codepoint it points
+/// to.  Can start anywhere in a stream of bytes.
```
Comment (clason) [inline src/nvim/mbyte.c:1841]:
```suggestion
/// to.  Can start anywhere in a stream of bytes.
/// Note: Counts individual codepoints of composed characters separately.
```

Comment (zeertzjq) [inline src/nvim/mbyte.c:1841]:
```suggestion
/// to.  Can start anywhere in a stream of bytes.
/// Composing characters are counted separately, not included in the character.
```

File: src/nvim/mbyte.c [hunk 5]
```diff
@@ -1837,10 +1837,11 @@ int mb_off_next(const char_u *base, const char_u *p)
...
+/// Composing characters are counted separately, not included in the character.
+/// Note: Counts individual codepoints of composed characters separately.
```
Comment (clason) [inline src/nvim/mbyte.c:1843]: One of these is enough; take your pick ;)

(My personal preference of course is the one that matches `utf_cp_head_off` below.)

Comment (dundargoc) [reply src/nvim/mbyte.c]: Missed that. Thanks.

**Comment by justinmk** [comment]
Doesn't look like we are using `cp` anywhere else, but I see that `utf_head_off` already exists so we need a disambiguiation. Possibly worth spelling out "codepoint" here unless we anticipate `cp` becoming more common.

**Comment by clason** [comment]
"Disambiguation" is the whole point of the exercise here. (Upstream introduced `utf_head_off` that counts _composed_ characters, not basic, uncomposed, codepoints which is needed for LSP.) 

The docstrings explicitly mention codepoints though?

**Review by justinmk — APPROVED** [review]

**Review by clason — APPROVED** [review]

**Pull request merged**